apollo-router/src/spec/schema.rs

```diff
@@ -63,53 +61,39 @@ impl Schema {
 
     pub(crate) fn parse(sdl: &str, configuration: &Configuration) -> Result<Self, SchemaError> {
         let start = Instant::now();
-        let mut compiler = ApolloCompiler::new();
-        let id = compiler.add_type_system(sdl, "schema.graphql");
-
-        let ast = compiler.db.ast(id);
+        let mut parser = apollo_compiler::Parser::new();
+        let ast = parser.parse_ast(sdl, "schema.graphql");
 
         // Trace log recursion limit data
-        let recursion_limit = ast.recursion_limit();
+        let recursion_limit = parser.recursion_reached();
```

Comment:
recursion_limit_reached?

Oh nvm it's the compiler's public api?

Reply:
it's the compiler api

Reply:
Yes [`recursion_reached`](https://docs.rs/apollo-compiler/1.0.0-beta.4/apollo_compiler/struct.Parser.html#method.recursion_reached) is from apollo-compiler. It says how much recursion was actually used/reached during the last parse, even if the limit was not exceeded. This can be useful to decide how high/low to configure the limit.

Reply:
thanks!